src/transports/WS.js

```diff
@@ -33,6 +35,10 @@ class WS extends Transport {
                 console.warn(error);
             }
         });
+
+        this.socket.addEventListener('error', error => {
+            throw new WebSocketError();
```

Comment:
This is asynchronous error. Please, provide special error handler in DeviceHive class for such errors